client/src/main/java/org/apache/celeborn/client/read/DfsPartitionReader.java

```diff
@@ -290,4 +290,15 @@ private void closeStream() {
   public PartitionLocation getLocation() {
     return location;
   }
+
+  @Override
+  public int getCurrentChunk() {
+    return returnedChunks;
+  }
+
+  @Override
+  public void setCurrentChunk(int currentChunk) {
```

Comment:
returnedChunks is not equal to currentChunkIndex, We shouldn't retry fetching chunk use returnedChunks. Because the process of fetching chunk is asynchronous, it is very likely that requests sent later will be received earlier.